File: algorithm/src/main/java/baekjoon/p11944/Main.java
package baekjoon.p11944;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt(), sc.nextInt()));
    }

    private static String solve(int n, int m) {
        StringBuilder sb = new StringBuilder();
        int repeat = n;

        while (repeat-- > 0) {
            sb.append(String.valueOf(n));
        }

        if (sb.length() > m) {
            return sb.substring(0, m);
        } else {
            return sb.toString();
        }
    }

}

File: algorithm/src/main/java/baekjoon/p11944/Main.kt
package baekjoon.p11944

fun main(args: Array<String>) {
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    println(solve(inputs[0], inputs[1]))
}

private fun solve(n: Int, m: Int): String {
    return n.toString().repeat(n).take(m)
}